src/reducers/secrets.js

```diff
@@ -14,14 +14,28 @@ limitations under the License.
 import { combineReducers } from 'redux';
 import merge from 'lodash.merge';
 import { ALL_NAMESPACES } from '@tektoncd/dashboard-utils';
-import { isStale } from '../utils';
+
+// TODO: replace isSecretStale() with utils.js isStale() when secrets store uses
+// byId structure.
+function isSecretStale(resource, state) {
+  const { name, namespace } = resource.metadata;
+  if (!state[namespace] || !state[namespace][name]) {
+    return false;
+  }
+  const existingVersion = parseInt(
+    state[namespace][name].metadata.resourceVersion,
+    10
+  );
+  const incomingVersion = parseInt(resource.metadata.resourceVersion, 10);
+  return existingVersion > incomingVersion;
+}
 
 function byNamespace(state = {}, action) {
   switch (action.type) {
     case 'SECRETS_FETCH_SUCCESS':
       const namespaces = action.data.reduce((accumulator, secret) => {
         const { metadata } = secret;
-        if (isStale({ metadata }, state, 'name')) {
```

Comment:
could we avoid creating a custom 'isStale' by adjusting the call to pass in the relevant slice of the state instead?

```js
const secretsState = state[namespace] || {};
if (isStale({ metadata }, secretsState, 'name')) {
```

Reply:
oh yeah this might be a better way  of doing it, I'll have a go at that now :)

Reply:
@AlanGreene works perfectly, thanks :) I've updated this now